docs/root/api-v1/http_filters/gzip_filter.rst

```diff
@@ -0,0 +1,70 @@
+.. _config_http_filters_gzip_v1:
+
+Gzip
+======
+
+Gzip :ref:`configuration overview <config_http_filters_gzip>`.
+
+.. code-block:: json
+
+  {
+    "name": "gzip",
+    "config": {
+      "memory_level": "...",
+      "content_length": "...",
+      "compression_level": "...",
+      "compression_strategy": "...",
+      "content_type": "...",
+      "disable_on_etag_header": "...",
+      "disable_on_last_modified_header": "...",
+      "window_bits": "...",
+      "disable_vary": "...",
+    }
+  }
+
+memory_level
+  *(optional, integer)* Value from 1 to 9 that controls the amount of internal memory used
+  by Zlib. Higher values use more memory, but are faster and produce better compression results.
+  The default value is 5.
+
+content_length
+  *(optional, integer)* Minimum response length, in bytes, which will trigger compression. The default
+  value is 30.
+
+compression_level
+  *(optional, string)* A value used for selecting Zlib's compression level. This setting will affect
+  speed and amount of compression applied to the content. "BEST" provides higher compression at the
+  cost of higher latency, "SPEED" provides lower compression with minimum impact on response time.
+  "DEFAULT" provides an optimal result between speed and compression. This field will be set
+  to "DEFAULT" if not specified.
+
+compression_strategy
+  *(optional, string)* A value used for selecting Zlib's compression strategy which is directly related
+  to the characteristics of the content. Most of the time "DEFAULT" will be the best choice, though
+  there are situations which changing this parameter might produce better results. For example,
+  Run-length encoding (RLE) is typically used when the content is known for having sequences which
+  same data occurs many consecutive times. For more information about each strategy, please refer to
```

Comment:
My expectation is that this setting wouldn't be adjusted much.  But if it were, it'd probably be better to express it as a map from content-type to strategy.

Reply:
That's a cool idea! In the case where it's empty, DEFAULT is applied for all content types. Let me go through this later.

👀

Reply:
I think most of the time this will be just set as DEFAULT.
For compressing PNG I can see that tweaking strategy could produce significantly better results -> http://www.libpng.org/pub/png/book/chapter07.html

I think this mapping is a great idea, but it should be included in a followup PR as a new setting instead. Thoughts?

Reply:
sounds good.